include/client.h

```diff
@@ -1026,6 +1051,132 @@ class Client
+        *   \throw SmartRedis::Exception if poll list length command fails
+        */
+        bool poll_list_length(const std::string& name, int list_length,
+                              int poll_frequency_ms, int num_tries);
+
+        /*!
+        *   \brief Get datasets from an aggregation list
+        *   \details The aggregation list key used to retrieve datasets
+        *            may be formed by applying a prefix to the supplied
+        *            name. See set_data_source() and use_list_ensemble_prefix()
+        *            for more details.  An empty or nonexistant
+        *            aggregation list returns an empty vector.
+        *   \param list_name The name of the aggregation list
+        *   \returns A vector containing DataSet objects.
+        *   \throw SmartRedis::Exception if retrieval fails.
+        */
+        std::vector<DataSet> get_datasets_from_list(const std::string& list_name);
+
+        /*!
+        *   \brief Get a range of datasets (by index) from an aggregation list
+        *   \details The aggregation list key used to retrieve datasets
+        *            may be formed by applying a prefix to the supplied
+        *            name. See set_data_source()  and use_list_ensemble_prefix()
+        *            for more details.  An empty or nonexistant aggregation
+        *            list returns an empty vector.  If the provided
+        *            end_index is beyond the end of the list, that index will
+        *            be treated as the last index of the list.  If start_index
+        *            and end_index are inconsistent (e.g. end_index is less
+        *            than start_index), an empty list of datasets will be returned.
+        *   \param list_name The name of the aggregation list
+        *   \param start_index The starting index of the range (inclusive,
+        *                      starting at zero).  Negative values are
+        *                      supported.  A negative value indicates offsets
+        *                      starting at the end of the list. For example, -1 is
+        *                      the last element of the list.
+        *   \param end_index The ending index of the range (inclusive,
+        *                    starting at zero).  Negative values are
+        *                    supported.  A negative value indicates offsets
+        *                    starting at the end of the list. For example, -1 is
+        *                    the last element of the list.
+        *   \returns A vector containing DataSet objects.
+        *   \throw SmartRedis::Exception if retrieval fails or
+        *          input parameters are invalid
+        */
+        std::vector<DataSet> get_dataset_list_range(const std::string& list_name,
+                                                    const int start_index,
+                                                    const int end_index);
```

Comment:
Rather than make the user calculate the range, should we mirror the approach in the multiGPU functions and have them provide the total number of processes plus their current rank -- and then provide the relevant "slice" of data?